vm/adb/adb.go

```diff
@@ -343,20 +344,27 @@ func (inst *instance) checkBatteryLevel() error {
 }
 
 func (inst *instance) getBatteryLevel(numRetry int) (int, error) {
-	out, err := inst.adb("shell", "dumpsys battery | grep level:")
+	var out, err = inst.doOneBatteryCheck()
 
 	// allow for retrying for devices that does not boot up so fast
-	for ; numRetry >= 0 && err != nil; numRetry-- {
+	for ; numRetry >= 0 && !(err == nil || strings.Compare(err.Error(), "non-battery device") == 0); numRetry-- {
 		if numRetry > 0 {
 			// sleep for 5 seconds before retrying
 			time.Sleep(5 * time.Second)
-			out, err = inst.adb("shell", "dumpsys battery | grep level:")
+			out, err = inst.doOneBatteryCheck()
+			// out, err = inst.adb("shell", "dumpsys battery | grep level:")
```

Comment:
remove dead code

Reply:
adb.go is reverted back to the original version. My edits are not needed. Adding "battery_check": false under vm works.